test/slider.test.tsx

```diff
@@ -0,0 +1,367 @@
+import { Page, test } from 'bigtest';
+import { Slider as Component } from '@material-ui/core';
+import { Slider, Thumb } from '../src/index';
+import { createRenderStep } from './helpers';
+
+const renderSlider = createRenderStep(
+  Component,
+  {
+    min: -100,
+    max: 200,
+    defaultValue: 0,
+    getAriaLabel: index =>
+      index == 0 ? '1st' :
+      index == 1 ? '2nd' :
+      index == 2 ? '3rd' :
+      'nth',
+    getAriaValueText: value => `${value}°C`
+  },
+  ({ props, children }) => (
+    <div style={{ width: '100px', height: '100px' }}>
+      <span id="label-id">Slider Label</span>
+      {children(props)}
+    </div>
+  )
+);
+const slider = Slider();
+const firstThumb = slider.find(Thumb('1st'));
+const secondThumb = slider.find(Thumb('2nd'));
+const thirdThumb = slider.find(Thumb('3rd'));
+const disabledSlider = Slider({ disabled: true });
+const disabledThumb = Thumb({ disabled: true });
+
+export default test("Slider")
+  .step(Page.visit("/"))
+  .child('Slider with one thumb', (test) => test
+    .child('default', (test) => test
+      .step(renderSlider())
+      .assertion(slider.exists())
+      .assertion(slider.has({ orientation: 'horizontal' }))
+      .assertion(slider.has({ value: 0 }))
+      .assertion(slider.has({ textValue: '0°C' }))
+      .assertion(slider.has({ minValue: -100 }))
+      .assertion(slider.has({ maxValue: 200 }))
+      .assertion(slider.is({ horizontal: true }))
+      .assertion(slider.is({ vertical: false }))
+      .assertion(slider.is({ min: false }))
+      .assertion(slider.is({ max: false }))
+      .assertion(slider.find(Thumb()).has({ index: 0 }))
+      .assertion(slider.find(Thumb()).has({ value: 0 }))
+      .assertion(slider.find(Thumb()).has({ textValue: '0°C' }))
+      .assertion(slider.find(Thumb('1st')).exists())
+      .assertion(slider.find(Thumb({ disabled: false })).exists())
+      .assertion(Slider('1st').exists())
+      .assertion(Slider({ disabled: false }).exists())
+      .assertion(Slider('My Slider').absent())
```

Comment:
Just out of curiosity, is there a reason you're not combining these assertions? (to make the tests look tidier (and to type less when you're writing the tests))
```js
.assertion(
  slider.has({ minValue: -100 }),
  slider.has({ maxValue: 200 }),
  slider.is({ horizontal: true }),
  slider.is({ vertical: false }),
  slider.is({ min: false }),
  slider.is({ mas: false }),
)
```

Reply:
I didn't know that I could write it like this ) thanks

Reply:
I didn't know that I could write it like this ) thanks